Doc update for CDS-97061- getinstance workaround (#6978)
harness/developer-hub: docs/continuous-delivery/manage-deployments/rollback-deployments.md

![](./static/permissions_post_rollback.png)
:::

> ## Trigger rollback for specific service using APIs

Author: ```suggestion
## Trigger rollback for specific services using APIs
```

You can use APIs to trigger rollback for specific services. For more details, go to [Trigger rollback](https://apidocs.harness.io/tag/Rollback/#operation/triggerRollback). 
Note that you need `instanceKey` and `infrastructureMappingId` of the service instance for rollback. Obtaining this info is not straigtforward. You can Inspect the Instance Details page to obtain these details as shown in the image below. 

> ![](./static/getinstances.png)

Reviewer: Instead of screenshot can we give the actual API with params? This way user does not have to go to UI

![](./static/permissions_post_rollback.png)
:::

> ## Trigger rollback for specific service using APIs

Author: ```suggestion
## Obtaining instance key and infrastructure mapping Id to roll back specific services using APIs
```

![](./static/permissions_post_rollback.png)
:::

## Trigger rollback for specific service using APIs

You can use APIs to trigger rollback for specific services. For more details, go to [Trigger rollback](https://apidocs.harness.io/tag/Rollback/#operation/triggerRollback). 

> Note that you need `instanceKey` and `infrastructureMappingId` of the service instance for rollback. Currently, obtaining this parameters is not straightforward. You can obtain these parameters using a cURL command. 

Author: ```suggestion
Note that you need `instanceKey` and `infrastructureMappingId` of the service instance for rollback. You can obtain these parameters using a cURL command. 
```

Here's a sample cURL request:

```

> curl --location 'https://localhost:8181/ng/api/dashboard/getInstancesDetails?routingId=<Account_Identifier>accountIdentifier=<Account_Identifier>orgIdentifier=default&projectIdentifier=Sarthak&serviceId=Kustomize&envId=Env&infraIdentifier=qatarget&buildId=null' \

Reviewer: can we add place holders for org, project, env, infra as well?

--header 'Accept: */*' \
--header 'Accept-Language: en-US,en;q=0.9' \

> --header 'Authorization: Bearer <YOUR_BEARER_TOKEN> \

Reviewer: For auth we can ask to use the x-api-key

![](./static/permissions_post_rollback.png)
:::

## Trigger rollback for specific services using APIs

You can use APIs to trigger rollback for specific services. For more details, go to [Trigger rollback](https://apidocs.harness.io/tag/Rollback/#operation/triggerRollback). 

> Note that you need `instanceKey` and `infrastructureMappingId` of the service instance for rollback. Obtaining this info is not straigtforward. You can Inspect the Instance Details page in the Harness UI to obtain these details from the `getinstancesDetails` API.

Reviewer 2: ```suggestion
Note that you need `instanceKey` and `infrastructureMappingId` of the service instance for rollback. Obtaining this info is not straight forward. You can Inspect the Instance Details page in the Harness UI to obtain these details from the `getinstancesDetails` API.
```
